plugins/logs/plugin_test.go

```diff
@@ -802,13 +804,35 @@ func TestPluginMasking(t *testing.T) {
 				"foo":          []map[string]interface{}{{"baz": 1}},
 				"password":     "mySecretPassword",
 			},
-			expected: map[string]interface{}{
+			expectedInput: map[string]interface{}{
 				"is_sensitive": true,
 				"jwt":          "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.KipSRURBQ1RFRCoq",
 				"bar":          1,
 				"foo":          []interface{}{map[string]interface{}{"changed": json.Number("1")}},
 			},
 		},
+		{
+			note: "complex result erase (deep copy)",
```

Comment:
Let's do this testing on the mask implementation in mask_test.go; we already have similar test cases there so I think we just have to modify `TestMaskRuleMask` in mask_test.go to check that the original result is unmodified.